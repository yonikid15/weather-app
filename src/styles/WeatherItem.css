.weatherItem {
  background:
    linear-gradient(
      rgba(0, 0, 0, 0.35), 
      rgba(0, 0, 0, 0.35)
    ),
    url('../../public/images/building-architecture-historical-tower.jpg') no-repeat center/cover;
  padding: 20px 0;
  margin: 0px 20px;
  border-radius: 10px;

  display: grid;
  grid-template-columns: repeat( auto-fill, 100px );
  grid-template-rows: repeat( auto-fill, 100px );
  justify-content: center;
  align-content: center;
}

.weatherItem:not(:first-child) {
  margin-top: 300px;
}