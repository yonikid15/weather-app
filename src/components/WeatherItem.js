import React from 'react';
import Place from './Place';
import Temperature from './Temperature';
import Humidity from './Humidity';
import Condition from './Condition';
import styles from '../styles/WeatherItem.css';
import backgroundImage from '../../public/images/nightsky1.jpeg';

const WeatherItem = ({ 
  id, 
  city, 
  country, 
  temperature, 
  humidity, 
  condition, 
  next, 
  prev, 
  active 
}) => {
  return (
    <div
      id={'activeItem'}
      className={ `${ styles.weatherItem }` } 
      style={{ 'backgroundImage' : `url(${ backgroundImage })` }}
      data-id={ id }
    >
      <Place 
        city={ city } 
        country={ country } 
      />
      <Temperature 
        temperature={ temperature }
      />
      <Humidity 
        humidity={ humidity }
      />
      <Condition 
        condition={ condition }
      />
    </div>
  );
};

export default WeatherItem;

