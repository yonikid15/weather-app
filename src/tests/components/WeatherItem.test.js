import React from 'react';
import { shallow } from 'enzyme';
import WeatherItem from '../../components/WeatherItem';

test('should render WeatherItem correctly', () => {
  const active = true;
  const data = {
    id: 0,
    city: 'Ottawa',
    country: 'CA',
    temperature: 300,
    humidity: 50,
    condition: 'clear skies',
    next: 1,
    prev: undefined
  }
  const wrapper = shallow( <WeatherItem active={ active } { ...data } /> );
  expect( wrapper ).toMatchSnapshot();
});