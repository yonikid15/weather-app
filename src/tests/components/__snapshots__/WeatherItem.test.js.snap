// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render WeatherItem correctly 1`] = `
<div
  className="weatherItem"
  data-id={0}
  id="activeItem"
  style={
    Object {
      "backgroundImage": "url([object Object])",
    }
  }
>
  <Place
    city="Ottawa"
    country="CA"
  />
  <Temperature
    temperature={300}
  />
  <Humidity
    humidity={50}
  />
  <Condition
    condition="clear skies"
  />
</div>
`;
